sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/CbsTokenProvider.cs

```diff
@@ -62,14 +83,94 @@ internal sealed class CbsTokenProvider : ICbsTokenProvider
         /// <param name="namespaceAddress">The address of the namespace to be authorized.</param>
         /// <param name="appliesTo">The resource to which the token should apply.</param>
         /// <param name="requiredClaims">The set of claims that are required for authorization.</param>
+        ///
         /// <returns>The token to use for authorization.</returns>
         ///
         public async Task<CbsToken> GetTokenAsync(Uri namespaceAddress,
                                                   string appliesTo,
-                                                  string[] requiredClaims)
+                                                  string[] requiredClaims) =>
+            Credential switch
+            {
+                _ when Credential.IsSharedAccessCredential => await AcquireSharedAccessTokenAsync().ConfigureAwait(false),
+                _ => await AcquireJwtTokenAsync().ConfigureAwait(false)
+            };
+
+        /// <summary>
+        ///   Performs the task needed to clean up resources used by the <see cref="CbsTokenProvider" />.
+        /// </summary>
+        ///
+        public void Dispose() => TokenAcquireSemaphore?.Dispose();
```

Comment:
I believe you can get away with not disposing of `SemaphoreSlim` if you don't access it's `AvailableWaitHandle` property. That way you wouldn't have to make this class `IDisposable`. @pakrym should be able to confirm if this is the case.

Reply:
https://stackoverflow.com/a/39195920/16249465

Reply:
I believe that you're right, but this class is used from a class which is also disposable;  I figured, no harm in following the "always call Dispose if a class has it" guidance since it's a non-invasive change.   That saves from a potential oversight if someone changes the use in the future but doesn't realize that we're not disposing.